src/subdomains/staking/application/services/staking.service.ts

```diff
@@ -94,6 +94,13 @@ export class StakingService {
     return this.repository.getByUserId(user.id);
   }
 
+  async getStakingsByUserId(userId: number): Promise<Staking[]> {
+    if(!userId) return [];
+    const user = await this.userService.getUserByKey('id', userId);
```

Comment:
I wouldn't use the `getUserByKey`, it's untyped and only for the support call 😉 there is a `getUser`method.